src/cpp/yaml/YamlReader.cpp

```diff
@@ -34,13 +28,13 @@ bool YamlReader::is_tag_present(
         const Yaml& yml,
         const TagType& tag)
 {
-    if (!yml.IsMap())
+    if (!yml.IsMap() && !yml.IsNull())
     {
         throw ConfigurationException(
                   utils::Formatter() << "Trying to find a tag: <" << tag << "> in a not yaml object map.");
     }
 
-    return (yml[tag]);
+    return yml[tag];
```

Comment:
Cast to bool.

Reply:
This seems to be a format warning more than a real one: https://stackoverflow.com/questions/206564/what-is-the-performance-implication-of-converting-to-bool-in-c

Checking the asm code generated for any of this forms, the result is the same. So I finally move to the easiest understable one: `?`

```
    int x;
    bool a;
    a = x;
    a = x ? true : false;
    a = !!x;
    a = (bool)x;
    a = static_cast<bool>(x);
```